server/with-bun/handlers/createWallet.ts

```diff
@@ -1,71 +1,67 @@
-  const token = authHeader.split(" ")[1];
-  const user = simulateVerifyToken(token);
+  const { email } = (await req.json()) as { email: string };
 
-  if (!user) {
-    return new Response("Unauthorized", { status: 401 });
+  if (!email) {
+    return new Response("Email is required in the request body", { status: 400 });
   }
 
-  // Parse the request body
-  const { email } = (await req.json()) as RequestBody;
-  if (user.email !== email) {
-    return new Response("Forbidden", { status: 403 });
-  }
-
-  // Ensure PARA_API_KEY is available
   const PARA_API_KEY = Bun.env.PARA_API_KEY;
   if (!PARA_API_KEY) {
-    return new Response("PARA_API_KEY not set", { status: 500 });
+    console.error("PARA_API_KEY environment variable is not set.");
+    return new Response("Server configuration error: PARA_API_KEY not set", { status: 500 });
   }
 
-  // Initialize Para client
-  const para = new ParaServer(Environment.BETA, PARA_API_KEY);
+  const para = new ParaServer(Environment.BETA, PARA_API_KEY, { disableWebSockets: true, disableWorkers: true });
 
-  // Check if a pre-generated wallet already exists
-  const hasPregenWallet = await para.hasPregenWallet({ pregenIdentifier: email, pregenIdentifierType: "EMAIL" });
-  if (hasPregenWallet) {
-    return new Response("Wallet already exists", { status: 400 });
-  }
+  try {
+    const hasPregenWallet = await para.hasPregenWallet({
+      pregenIdentifier: email,
+      pregenIdentifierType: "EMAIL",
+    });
+    if (hasPregenWallet) {
+      console.warn(`Wallet creation attempt for existing email: ${email}`);
+      return new Response(`Wallet already exists for ${email}`, { status: 409 });
+    }
 
-  // Create a new wallet
-  const wallet = await para.createPregenWallet({
-    type: WalletType.EVM,
-    pregenIdentifier: email,
-    pregenIdentifierType: "EMAIL",
-  });
-  if (!wallet) {
-    return new Response("Failed to create wallet", { status: 500 });
-  }
+    const wallet = await para.createPregenWallet({
+      type: WalletType.EVM,
+      pregenIdentifier: email,
+      pregenIdentifierType: "EMAIL",
+    });
+    if (!wallet) {
+      console.error(`Para SDK returned no wallet object for ${email}, but no error was thrown.`);
+      return new Response("Failed to create wallet (unknown reason)", { status: 500 });
+    }
 
-  // Retrieve and encrypt the key share
-  const keyShare = para.getUserShare();
-  if (!keyShare) {
-    return new Response("Failed to get key share", { status: 500 });
-  }
+    const keyShare = para.getUserShare();
+    if (!keyShare) {
+      console.error(`Failed to get key share immediately after creating wallet for ${email}`);
+      return new Response("Failed to retrieve key share after wallet creation", { status: 500 });
+    }
 
-  const encryptedKeyShare = encrypt(keyShare);
+    let encryptedKeyShare: string;
 
-  // Store the encrypted key share in the database
-  await setKeyShareInDB(email, encryptedKeyShare);
+    try {
```

Comment:
looking at this, it's a good amount of implementation. Could we pull more of this functionality around error responses into the SDK?